### PR TITLE
Add customized jenkinsfile

### PR DESCRIPTION
**Pull request by sladyn98**
This PR adds a react build step and also maven options

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -28,6 +37,17 @@ for (int i = 0; i < platforms.size(); ++i) {
                     }
                 }
 
+                stage('React Build') {
```
Comment (kwhetstone) [inline Jenkinsfile:40]: The problem here is that you're mixing Jenkinsfile conventions.  The agent syntax comes from declarative pipelines while everything else is scripted.  There are 2 paths to go down here: either convert the pipeline to declarative, or use a sidecar agent to launch and test this <https://www.jenkins.io/doc/book/pipeline/docker/#advanced-usage-with-scripted-pipeline>

I'm not sure if there's a node image in the repo, so I think you might be able to just use the alpine image.

Comment (sladyn98) [reply Jenkinsfile:40]: Yeah I guess we could go with a declarative syntax.

Comment (sladyn98) [reply Jenkinsfile:35]: I guess the node example was given on the website examples so don't you think it will run ?

Comment (kwhetstone) [reply Jenkinsfile:35]: The examples are for declarative pipelines not scripted like you're using here.  You can click through on the red x below to see what's failing.  It looks like there are still issues with your file

File: Jenkinsfile [hunk 2]
```diff
@@ -1,46 +1,60 @@
...
-                    junit '**/target/surefire-reports/TEST-*.xml'
-
-                    if (label == 'linux') {
-                      archiveArtifacts artifacts: '**/target/**/*.jar'
-                      findbugs pattern: '**/target/findbugsXml.xml'
-                    }
+        stage('React Build') {
+            agent {
+            docker {
+                image 'node:6-alpine'
```
Comment (kwhetstone) [inline Jenkinsfile:43]:
```suggestion
                image 'node:6'
```

**Pull request merged**